src/dbnode/persist/fs/commitlog/iterator.go

```diff
@@ -202,20 +191,14 @@ func (i *iterator) nextReader() bool {
 	return true
 }
 
-func filterFiles(opts Options, files []string, predicate FileFilterPredicate) ([]string, error) {
-	filteredFiles := make([]string, 0, len(files))
-	var multiErr xerrors.MultiError
+func filterFiles(opts Options, files []File, predicate FileFilterPredicate) []File {
+	filteredFiles := make([]File, 0, len(files))
 	for _, file := range files {
-		start, duration, _, err := ReadLogInfo(file, opts)
-		if err != nil {
-			multiErr = multiErr.Add(err)
-			continue
-		}
-		if predicate(file, start, duration) {
+		if predicate(file.FilePath, file.Start, file.Duration) {
```

Comment:
Would it be better to make predicate just take `commitlog.File` so it's easier to add new fields in the future?

Reply:
good call